### PR TITLE
User can view product details (eh-product-details2)

### PR DESCRIPTION
**Pull request by elliot-huck**
:
- **Purpose of PR**
This PR enables a user to view the details of any product, including how many remain unordered. It also includes a currently non-functional "Add to Order" button that is disabled if there are no available products to be ordered.

- **Issue number this request impacts** ------
Completes issue #1 

- **Description of what was changed:**
1. Added scaffolded ProductController and Product views.
1. Changed the Details controller and the Details view.
1. Added ProductDetailTest to BangazonTest directory

- **Thorough list of steps to test**
```
git fetch -a
git checkout eh-product-details2
start Bangazon.sln
```
- [x] Build the project and view it in the browser
- [x] Navigate to the /Products URL
- [x] Click the Details link for any Product to view product details. If you're using just the seed data:
- - [x] All products except product 1 (the Samsung TV) should have a blue "Add to Order" button. The button does not currently add anything to an order, but it should redirect the user to the Index page of all products
- - [x] The Samsung TV should have 0 remaining and the "Add to Order" button should be grayed out and disabled
- - If your database is not the same as the seed data, INSERT a few records into the OrderProduct table to make sure that all products show the number of items remaining, not quantity originally posted. Also, make sure you add enough intersections to the OrderProduct table that at least one product is reduced to 0 remaining to make sure the "Add to Order button is disabled
- [x] Open the Test Explorer in Visual Studio
- - From the top menu ribbon, go to Test > Windows > Test Explorer
- - Or use the keyboard shortcut: CTRL + E then T
- [x] Make sure that the ProductDetailTest shows up in the window.
- [x] Run all the tests by either clicking "Run All" in the Test Explorer window or running `dotnet test` in your console to make sure that all tests pass

if you don't do this you will be
**FIRED**

- **Checklist:** 
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added test instructions that prove my fix is effective or that my feature works
- [x] Run test suite with ```dotnet test``` and making sure all tests are green.

### CODE REVIEW

File: Bangazon/Views/Products/Edit.cshtml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+﻿@model Bangazon.Models.Product
```
Comment (Shu52) [inline Bangazon/Views/Products/Edit.cshtml]: This file is outside scope of the ticket. May cause future merge conflicts.

File: Bangazon/Views/Products/Delete.cshtml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿@model Bangazon.Models.Product
```
Comment (Shu52) [inline Bangazon/Views/Products/Delete.cshtml]: This file is outside scope of the ticket. May cause future merge conflicts.

File: Bangazon/Views/Products/Create.cshtml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+﻿@model Bangazon.Models.Product
```
Comment (Shu52) [inline Bangazon/Views/Products/Create.cshtml]: This file is outside scope of the ticket. May cause future merge conflicts.

File: Bangazon/Views/Products/Edit.cshtml (hunk 1 above)
Comment (elliot-huck) [reply Bangazon/Views/Products/Edit.cshtml]: I commented out unused methods on the scaffolded controller, but I left the views "as is"

File: Bangazon/Views/Products/Delete.cshtml (hunk 2 above)
Comment (elliot-huck) [reply Bangazon/Views/Products/Delete.cshtml]: I commented out unused methods on the scaffolded controller, but I left the views "as is"

File: Bangazon/Views/Products/Create.cshtml (hunk 3 above)
Comment (elliot-huck) [reply Bangazon/Views/Products/Create.cshtml]: I commented out unused methods on the scaffolded controller, but I left the views "as is"

**Pull request merged**